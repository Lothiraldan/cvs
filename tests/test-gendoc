#!/bin/sh

"$TESTDIR/hghave" rst2html || exit 80
RST2HTML=`which rst2html 2> /dev/null || which rst2html.py`

HGENCODING=UTF-8
export HGENCODING

for PO in C $TESTDIR/../i18n/*.po; do
    LOCALE=`basename $PO .po`
    echo
    echo "% extracting documentation from $LOCALE"
    echo ".. -*- coding: utf-8 -*-" > gendoc-$LOCALE.txt
    echo "" >> gendoc-$LOCALE.txt
    LC_ALL=$LOCALE python $TESTDIR/../doc/gendoc.py >> gendoc-$LOCALE.txt || exit

    # We run rst2html over the file without adding "--halt warning" to
    # make it report all errors instead of stopping on the first one.
    echo "checking for parse errors with rst2html"
    $RST2HTML gendoc-$LOCALE.txt /dev/null
done
