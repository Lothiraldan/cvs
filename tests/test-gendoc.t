#require docutils
#require gettext

Test document extraction

  $ HGENCODING=UTF-8
  $ export HGENCODING
  $ { echo C; ls "$TESTDIR/../i18n"/*.po | sort; } | while read PO; do
  >     LOCALE=`basename "$PO" .po`
  >     echo "% extracting documentation from $LOCALE"
  >     LANGUAGE=$LOCALE python "$TESTDIR/../doc/gendoc.py" >> gendoc-$LOCALE.txt 2> /dev/null || exit
  > 
  >     if [ $LOCALE != C ]; then
  >         if [ ! -f $TESTDIR/test-gendoc-$LOCALE.t ]; then
  >             echo missing test-gendoc-$LOCALE.t
  >         fi
  >         cmp -s gendoc-C.txt gendoc-$LOCALE.txt && echo "** NOTHING TRANSLATED ($LOCALE) **"
  >     fi
  > done; true
  % extracting documentation from C
  % extracting documentation from da
  % extracting documentation from de
  % extracting documentation from el
  % extracting documentation from fr
  % extracting documentation from it
  % extracting documentation from ja
  % extracting documentation from pt_BR
  % extracting documentation from ro
  % extracting documentation from ru
  % extracting documentation from sv
  % extracting documentation from zh_CN
  % extracting documentation from zh_TW
