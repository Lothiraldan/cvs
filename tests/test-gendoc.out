checking for syntax errors in gendoc.py
checking for parse errors with rst2html
